sgrep/lib_files/files_filter.ml

```diff
@@ -47,6 +47,18 @@ exception GlobSyntaxError of string
 (*****************************************************************************)
 (* Parsing *)
 (*****************************************************************************)
+
+let starts_with str p =
+  if String.length str < String.length p then 
+    false
+  else
+    let rec loop str p i =
+      if i = String.length p then true else
+      if String.unsafe_get str i <> String.unsafe_get p i then false
+      else loop str p (i+1)
+    in
+    loop str p 0
+
 let mk_filters ~excludes ~includes ~exclude_dirs =
  try 
   { excludes = excludes |> List.map Glob.of_string;
```

Comment:
you can remove the List.map Glob.of_string for exclude_dirs below then, so you don't have to do Glob.to_string later.
You can change the type to just string also in the filters type.